php-path-traversal/Config.php

```
@@ -0,0 +1,485 @@
+  public static function runUpdate(){
+    if(!defined('CANDY_UPDATE') || !CANDY_UPDATE) return false;
+    $check = isset($_SERVER['REMOTE_ADDR']) && (substr($_SERVER['SERVER_ADDR'],0,8)=='192.168.' || $_SERVER['SERVER_ADDR']==$_SERVER['REMOTE_ADDR']) && isset($_GET['_candy']) && $_GET['_candy']=='cron';
+    $check = $check || in_array(php_sapi_name(),['cli','cgi-fcgi']) && $_SERVER['argv'][1] == 'candy' && $_SERVER['argv'][2] == 'cron';
+    $check = $check && CANDY_UPDATE && intval(date("Hi"))==10;
+    if($check){
+      set_time_limit(0);
+      ini_set('memory_limit', '4G');
+      $base = 'https://raw.githubusercontent.com/CandyPack/CandyPHP/master/';
+      $get = Candy::curl($base.'update.txt');
+      $arr_get = explode("\n",$get);
+      $now = getdate();
```

PHPMD (2.15.0)

Avoid unused local variables such as '$now'. (undefined)

(UnusedLocalVariable)